### PR TITLE
DOP-4645: Internal Page Nav - Dark Mode

### PR DESCRIPTION
**Pull request by mmeigs**
### Stories/Links:

DOP-4645

### Current Behavior:

[Atlas](https://www.mongodb.com/docs/atlas/tutorial/create-atlas-account/)

### Staging Links:

[Atlas](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/matt.meigs/DOP-4645-darkmode-nav/tutorial/create-atlas-account/index.html)

### Notes:

Changes arrow color and underline color according to design's input in [this conversation](https://mongodb.slack.com/archives/C076Z47AFTL/p1717770500793109).

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: src/components/InternalPageNav.js [hunk 1]
```diff
@@ -23,11 +24,12 @@ const StyledContainer = styled.div`
...
   line-height: 28px;
   align-content: center;
-  color: ${palette.black};
+  color: var(--arrow-color);
 `;
 
 const titleSpanStyling = css`
   line-height: 28px;
+  --hover-text-decoration-color: var(--underline-color) !important;
 `;
```
Comment (mmeigs) [inline src/components/InternalPageNav.js:33]: I do hate using `!important`. But I thought this was the cleanest of the options. This color has to override the `Link`'s inline style... so it's either this or pass a prop down to `Link` simply to change the hover color.

File: src/components/InternalPageNav.js [hunk 2]
```diff
@@ -36,11 +38,17 @@ const LinkContentContainer = styled.div`
...
+  const { darkMode } = useDarkMode();
   const slugIndex = toctreeOrder.indexOf(slug);
   const prevSlug = slugIndex > 0 ? toctreeOrder[slugIndex - 1] : null;
   const nextSlug = slugIndex < toctreeOrder.length - 1 ? toctreeOrder[slugIndex + 1] : null;
   return (
-    <StyledContainer>
+    <StyledContainer
+      style={{
+        '--arrow-color': darkMode ? palette.gray.light2 : palette.black,
+        '--underline-color': darkMode ? '#5C6C75' : palette.gray.light2,
```
Comment (mayaraman19) [inline src/components/InternalPageNav.js:49]: The color we're using here is `#5C6C75` vs `palette.gray.dark1` is `#5c6d75`. probably a question for design but is it possible to use the latter just so we can avoid hardcoding?

**Review by mayaraman19 — APPROVED** [review]
Looks good, left a tiny comment below

File: src/components/InternalPageNav.js (hunk 2 above)
Comment (mmeigs) [reply src/components/InternalPageNav.js:49]: I'll ask Liz!

Comment (mmeigs) [reply src/components/InternalPageNav.js:49]: Good catch! She agrees with you!

**Pull request merged**